src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -935,7 +973,9 @@ public enum JsonSourceGenerationMode
     }
     public partial class JsonStringEnumConverter : System.Text.Json.Serialization.JsonConverterFactory
     {
+        [System.Diagnostics.CodeAnalysis.RequiresDynamicCode("JSON serialization and deserialization might require types that cannot be statically analyzed and might need runtime code generation. Use System.Text.Json source generation for native AOT applications.")]
         public JsonStringEnumConverter() { }
+        [System.Diagnostics.CodeAnalysis.RequiresDynamicCode("JSON serialization and deserialization might require types that cannot be statically analyzed and might need runtime code generation. Use System.Text.Json source generation for native AOT applications.")]
```

Comment:
It is rather unfortunate that we can't support converting enums to and from strings without requiring dynamic code. It feels like we should be able to make something work here for NativeAOT.

I don't think you need to solve this in this PR, so maybe we can open an issue for it.

Reply:
> It is rather unfortunate that we can't support converting enums to and from strings without requiring dynamic code.

Runtime allows you to convert enums to and from strings with NativeAOT. The problem is that the EnumConverter implementation creates new generic types dynamically: `typeof(EnumConverter<>).MakeGenericType`.

Reply:
Right. I'm thinking (in a separate PR) we add a switch to JsonStringEnumConverter on `RuntimeFeature.IsDynamicCodeSupported`, and fallback to some other implementation when dynamic code isn't supported. Then we can remove this attribute and support this API in NativeAOT.

Reply:
Is the `JsonStringEnumConverter` used with serialization source generators?

If yes, I think there should be a static path that is both fast and that does not require dynamical code generation. For example, can `EnumConverter<T>` be public and source generators changed to use it?

If not, I do not think we need to bother with making it AOT friendly.

Reply:
@eiriktsarpalis @layomia @krwq - can one of you answer this? Can the `JsonStringEnumConverter` be used outside of the reflection-based serialization? i.e. with source generation.